Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesEmbeddedTrait.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  AppcuesEmbeddedTrait.swift
+//  AppcuesKit
+//
+//  Created by Matt on 2023-06-13.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+internal class AppcuesEmbeddedTrait: AppcuesStepDecoratingTrait, AppcuesWrapperCreatingTrait, AppcuesPresentingTrait {
+
+    struct Config: Decodable {
+        let frameID: String
+        let style: ExperienceComponent.Style?
+        let transition: AppcuesFrameView.Transition?
+    }
+
+    static let type = "@appcues/embedded"
+
+    weak var metadataDelegate: AppcuesTraitMetadataDelegate?
+
+    private weak var appcues: Appcues?
+
+    private let frameID: String
+    private let transition: AppcuesFrameView.Transition
+    private let style: ExperienceComponent.Style?
+
+    // The view embedded in the customer application
+    weak var embedView: AppcuesFrameView?
+
+    required init?(configuration: AppcuesExperiencePluginConfiguration) {
+        self.appcues = configuration.appcues
+
+        guard let config = configuration.decode(Config.self) else { return nil }
+        self.frameID = config.frameID
+        self.transition = config.transition ?? .none
+        self.style = config.style
+    }
+
+    func decorate(stepController: UIViewController) throws {
+        // Need to cast for access to the padding property.
+        guard let stepController = stepController as? ExperienceStepViewController else { return }
+        stepController.padding = NSDirectionalEdgeInsets(paddingFrom: style)
+    }
+
+    func createWrapper(around containerController: AppcuesExperienceContainerViewController) throws -> UIViewController {
+        applyStyle(style, to: containerController)
+        return containerController
+    }
+
+    func addBackdrop(backdropView: UIView, to wrapperController: UIViewController) {
+        // no backdrop on embeds
+    }
+
+    func present(viewController: UIViewController, completion: (() -> Void)?) throws {
+        let experienceRenderer = appcues?.container.resolve(ExperienceRendering.self)
+
+        self.embedView = experienceRenderer?.owner(forContext: .embed(frameID: frameID)) as? AppcuesFrameView
```

Comment:
following along - the key spot where the registered view is looked up at render time